### PR TITLE
Qualcomm AI Engine Direct - Fix Full

### PR DESCRIPTION
**Pull request by winskuo-quic**
### Summary
Based on reported issue: https://github.com/pytorch/executorch/pull/19179
2 mathematically equivalent model in nn.Module lowered both lowered to QNN, however, 1 accuracy is good while another is bad. Debugged with qnn_intermediate_debugger and targeted that accuracy drop is possibly caused by `full` operation. Took a look and noticed the op builder did not take layout transform into consideration.
image

### Test plan
Passing tests under test_qnn_delegate.py and fix issue in: https://github.com/pytorch/executorch/pull/19179

### CODE REVIEW

**Review by digantdesai — APPROVED** [review]
Any test to prevent future regression?

**Comment by winskuo-quic** [comment]
> Any test to prevent future regression?

@digantdesai Thanks for reviewing the PR. I have added some UTs that will fail without this change and passes with this PR.

**Comment by abhinaykukkadapu** [comment]
Thanks for the fix @winskuo-quic

**Review by abhinaykukkadapu — APPROVED** [review]
Thanks @winskuo-quic

**Review by psiddh — APPROVED** [review]
